vhost-device-sound/src/audio_backends/pipewire.rs

```diff
@@ -208,6 +208,16 @@ impl AudioBackend for PwBackend {
 
             let params = &stream_params[stream_id as usize].params;
 
+            if let Some(stream) = stream_hash.get(&stream_id) {
+                let disconnect_result = stream.disconnect();
+                stream_hash.remove(&stream_id);
+                stream_listener.remove(&stream_id);
+                if let Err(err) = disconnect_result {
```

Comment:
No real reason to bind the disconnect result. Just a nitpick and not a functional change, otherwise LGTM.

```suggestion   
                stream_hash.remove(&stream_id);
                stream_listener.remove(&stream_id);
                if let Err(err) = stream.disconnect() {
```

Reply:
Here I think the order of operation is important, I want to disconnect the stream before removing it from the hashmap and stream_listener. so I'd prefer to leave it as it is.

Reply:
@dorindabassey IIUC we acquired the write lock for both, so why the order is important?

Reply:
(It isn't important, and since those statements don't seem to have side effects to each other the compiler is actually within rights to re-order them as it pleases. Take this with a pinch of salt because I didn't actually check if that happens in this code.)

Reply:
going by your suggestion @epilys the borrow checker complains ```error[E0502]: cannot borrow `stream_hash` as mutable because it is also borrowed as immutable
   --> vhost-device-sound/src/audio_backends/pipewire.rs:212:17
    |
211 |             if let Some(stream) = stream_hash.get(&stream_id) {
    |                                   ----------- immutable borrow occurs here
212 |                 stream_hash.remove(&stream_id);
    |                 ^^^^^^^^^^^ mutable borrow occurs here
213 |                 stream_listener.remove(&stream_id);
214 |                 if let Err(err) = stream.disconnect() {
    |                                   ------ immutable borrow later used here
``` which was why i left it this way.